File: java/src/main/java/com/android/util/ByteUnit.java
package com.android.util;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Created by xuzhb on 2019/12/8
 * Desc:流量的单位
 */
public class ByteUnit {

    public static final String B = "byte";
    public static final String KB = "kb";
    public static final String MB = "mb";
    public static final String GB = "gb";

    @StringDef({B, KB, MB, GB})
    @Retention(RetentionPolicy.SOURCE)
    public @interface ByteUnitDef {
    }

    private static final double UNIT = 1024;

    public static double convertByteUnit(double totalBytes, @ByteUnitDef String unit) {
        if (totalBytes > 0) {
            if (ByteUnit.KB.equals(unit)) {
                totalBytes /= Math.pow(UNIT, 1);
            } else if (ByteUnit.MB.equals(unit)) {
                totalBytes /= Math.pow(UNIT, 2);
            } else if (ByteUnit.GB.equals(unit)) {
                totalBytes /= Math.pow(UNIT, 3);
            }
        }
        return totalBytes;
    }

}

File: kotlin/src/main/java/com/android/util/ByteUnit.kt
package com.android.util

import androidx.annotation.StringDef
import kotlin.math.pow

/**
 * Created by xuzhb on 2020/12/1
 * Desc:流量的单位
 */
object ByteUnit {

    const val B = "byte"
    const val KB = "kb"
    const val MB = "mb"
    const val GB = "gb"

    @StringDef(B, KB, MB, GB)
    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    annotation class ByteUnitDef

    private const val UNIT = 1024.0

    fun convertByteUnit(totalBytes: Double, @ByteUnitDef unit: String): Double {
        var totalBytes = totalBytes
        if (totalBytes > 0) {
            when (unit) {
                KB -> totalBytes /= UNIT.pow(1.0)
                MB -> totalBytes /= UNIT.pow(2.0)
                GB -> totalBytes /= UNIT.pow(3.0)
            }
        }
        return totalBytes
    }

}